src/manager/checkpoint/subsystem.rs

```diff
@@ -179,12 +188,16 @@ async fn submit_till_current_epoch(manager: &impl CheckpointManager) -> Result<(
     );
 
     let mut next_epoch = last_executed_epoch + period;
-    while next_epoch < current_epoch {
+    let cut_off_epoch = std::cmp::min(
+        current_epoch,
+        SUBMISSION_LOOK_AHEAD_EPOCH + last_executed_epoch,
+    );
+    while next_epoch < cut_off_epoch {
         // now we process each validator
         for validator in &validators {
             log::debug!("submit checkpoint for validator: {validator:?} in manager: {manager:}");
 
-            if manager
+            if !manager
```

Comment:
Would you mind explaining this and the `cut_off_epoch` (I may understand it better in the next PR, but just in case I forget).

Reply:
I put a comment in the PR as well, just paste it here for reference:
```
Instead of loop all the way to `current_epoch`, we loop till `cut_off_epoch`.
Reason because if the current epoch is significantly greater than last_executed_epoch and there are lots of validators in the network, loop all the way to current epoch might have some outdated data. Set a cut off epoch such that validators can sync with chain more regularly.
```